### PR TITLE
WIP: Custom DNS changes for AWS

### PR DESCRIPTION
**Pull request by sadasu**

### CODE REVIEW

File: pkg/terraform/stages/aws/stages.go [hunk 1]
```diff
@@ -41,3 +52,37 @@ func customBootstrapDestroy(s stages.SplitStage, directory string, terraformDir
 		"failed to destroy bootstrap",
 	)
 }
+
+func extractAWSLBConfig(s stages.SplitStage, directory string, config *types.InstallConfig) ([]string, []string, error) {
```
Comment (barbacbd) [inline pkg/terraform/stages/aws/stages.go:56]: If the user is going to supply load balancer (internal and external) information, why do we need to extract it from terraform? Shouldn't terraform skip any creation of this data, so we would already know this information and can populate any structure without the output of terraform right ?

File: pkg/terraform/stages/aws/stages.go [hunk 2]
```diff
@@ -41,3 +52,37 @@ func customBootstrapDestroy(s stages.SplitStage, directory string, terraformDir
...
+	var apiLBDNSNames []string
+	if apiLBDNSNameRaw, ok := outputs["aws_external_api_lb_dns_name"]; ok {
+		apiLBDNSNames[0], ok = apiLBDNSNameRaw.(string)
+		if !ok {
+			return nil, nil, errors.New("could not read API LB DNS Name from terraform outputs")
+		}
+	}
+
+	var apiIntLBDNSNames []string
+	if apiLBDNSNameRaw, ok := outputs["aws_external_api_lb_dns_name"]; ok {
```
Comment (barbacbd) [inline pkg/terraform/stages/aws/stages.go:81]: Looks like a copy/paste here, probably want the internal name.

File: pkg/terraform/stages/aws/stages.go (hunk 1 above)
Comment (sadasu) [reply pkg/terraform/stages/aws/stages.go:56]: We donot expect the user to provide LB information anymore. They will not pre-create these resources. Instead, Installer will create these resources like a regular IPI install and only skip route53 configuration.

**Comment by sadasu** [comment]
/close

AWS implementation should be based on the CAPI work.

**Pull request closed without merging**